### ISSUE TITLE
3.23.0 seems to have broken something

### ISSUE THREAD

**Issue by TD-4242** [Flagsmith/flagsmith-js-client#217]
Our test environments silently stopped working this morning.  only thing that fixed it was rolling back flagsmith client version.

I've done some digging and can't seem to find anything wrong.  Everything looks like it inits, I can run all the flagsmith functions and get the expected results from the browser console, but something is locking up and not allowing my application to continue.

rolling back to 3.22.1 fixes it
swapping back and forth is 100% reproducible

**Comment by TD-4242** [comment]
This seems like it may be the onChange function.  still troubleshooting.

**Comment by novakzaballa** [comment]
Thank you for reporting this issue. Could you please let us know which framework are you using and which version (i.e. React v18)?

Is it possible and feasible for you to provide us with a minimal but complete project reproducing the issue?

**Comment by TD-4242** [comment]
May take a bit to create a full project, here is some pseudo code the encapsulates the core of how we load flagsmith along with some of my debugging:

```bash
grep flagsmith package.json
"flagsmith": "3.22.1", #works
grep flagsmith package.json
"flagsmith": "3.23.0", #doesnt work

```

```typescript
const onChange = () => {
  console.log('flagsmith onChange');
  loadApp(); // load the app after flagsmith is initialized
};

console.log('pre flagsmith init');
flagsmith.init({
  api: flagsmith_url,
  cacheFlags: true,
  defaultFlags,
  environmentID: flagsmith_key ?? 'Not Defined',
  enableAnalytics: true,
  identity: userId,
  onChange, // expected to be called on init
  enableLogs: true,
  onError: async () => {
    console.log('flagsmith error');
    window.logger.error('flagsmith_init', {
      message: 'error initializing flagsmith url',
    });
  },
});
console.log('post flagsmith init');

setTimeout(onChange, 10000) // call onChange after 10 seconds as a debug to see if it's calle
```

**Comment by kyle-ssg** [comment]
Hi @TD-4242 thanks for raising and apologies for the issue, I've replicated your issue. This should be fixed in 3.23.2, please let me know if this is still an issue and we'll investigate further.
